hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/jdbc/JdbcJoinFullScanProcessorSupplier.java

```diff
@@ -0,0 +1,168 @@
+import com.hazelcast.jet.impl.processor.TransformP;
+import com.hazelcast.jet.sql.impl.ExpressionUtil;
+import com.hazelcast.jet.sql.impl.JetJoinInfo;
+import com.hazelcast.jet.sql.impl.connector.jdbc.util.ResultSetUtils;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.DataSerializable;
+import com.hazelcast.security.impl.function.SecuredFunction;
+import com.hazelcast.security.permission.ConnectorPermission;
+import com.hazelcast.sql.impl.expression.Expression;
+import com.hazelcast.sql.impl.expression.ExpressionEvalContext;
+import com.hazelcast.sql.impl.row.JetSqlRow;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.security.Permission;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.IntStream;
+
+import static com.hazelcast.jet.Traversers.traverseIterable;
+import static com.hazelcast.security.permission.ActionConstants.ACTION_READ;
+import static java.util.Collections.singletonList;
+import static java.util.stream.Collectors.toList;
+
+public class JdbcJoinFullScanProcessorSupplier
+        extends AbstractJdbcSqlConnectorProcessorSupplier
+        implements DataSerializable, SecuredFunction {
+
+    private String selectQuery;
+
+    private JetJoinInfo joinInfo;
+
+    private List<Expression<?>> projections;
+
+    // Transient members are received when ProcessorSupplier is initialized.
+    // No need to serialize them
+
+
+    private transient ExpressionEvalContext expressionEvalContext;
+
+
+    // Classes conforming to DataSerializable should provide a no-arguments constructor.
+    public JdbcJoinFullScanProcessorSupplier() {
+    }
+
+    public JdbcJoinFullScanProcessorSupplier(@Nonnull NestedLoopReaderParams nestedLoopReaderParams,
+                                             @Nonnull String selectQuery) {
+        super(nestedLoopReaderParams.getJdbcTable().getDataConnectionName());
+        this.selectQuery = selectQuery;
+        this.joinInfo = nestedLoopReaderParams.getJoinInfo();
+        this.projections = nestedLoopReaderParams.getProjections();
+    }
+
+    @Override
+    public void init(@Nonnull Context context) throws Exception {
+        super.init(context);
+        this.expressionEvalContext = ExpressionEvalContext.from(context);
+    }
+
+    @Nonnull
+    @Override
+    public Collection<? extends Processor> get(int count) {
+        FunctionEx<JetSqlRow, Traverser<JetSqlRow>> joinFunction = createJoinFunction(
+                dataConnection,
+                selectQuery,
+                joinInfo,
+                expressionEvalContext,
+                projections);
+        return IntStream.range(0, count)
+                .mapToObj(i -> new TransformP<>(joinFunction)).
```

Comment:
This will call the database for each item. We should use `TransformBatchedP` and call it with a batch of items.

Reply:
for that you will have to generate query with `IN` instead of `=` or series of `OR` / `AND` statements when there is more than one condition. Afterwards you will have to match rows from DB with rows from Jet. This is all doable (we do it for IMap) but complex.

Reply:
rev : b9bf0564bfd61ffc61d627916532a49e0462d633 
This commit is using TransformBatchedP . Also added more test cases

Reply:
This still executes the query one by one for each row. I meant what @k-jamroz describes above - collect a batch of left rows, and generate a query with `IN` clause to get the results in a single query (saving round trips to the database).

Reply:
Added "union all" to SQLs to execute all queries in one go. The received Resultset has an extra column that shows which "leftrownumber" it belongs to. Using this column it is possible to associate the leftRow with Resultset . 

Also added unit tests for cross join, self join, theta join. And some tests for MySQL + Postgres